src/engine/runtime.js

```diff
@@ -439,6 +439,11 @@ Runtime.prototype.startHats = function (requestedHatOpcode,
     }
     var instance = this;
     var newThreads = [];
+
+    for (var opts in optMatchFields) {
+        optMatchFields[opts] = optMatchFields[opts].toUpperCase();
+    }
```

Comment:
Doh — I thought I made sure it wasn't going to have a performance effect to do it differently. Can I blame no coffee? I think it's fine as-is in that case.